slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -19,9 +19,45 @@ std::atomic<bool> b_continue_session{true};
 ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
                                             const GetPositionRequest *request,
                                             GetPositionResponse *response) {
-    LOG(ERROR) << "GetPosition is not yet implemented.\n";
-    return grpc::Status(grpc::StatusCode::UNIMPLEMENTED,
-                        "GetPosition is not yet implemented.");
+
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        if (map_builder.GetLocalSlamResultPoses().size() > 0) {
```

Comment:
I suggest we save the latest global pose as a member variable within the slam service and update it as we're building the map continuously. Then, we can use that member variable here.